src/main/java/com/google/cloud/tools/appengine/cloudsdk/CloudSdk.java

```diff
@@ -59,6 +59,7 @@
   private static final String DEV_APPSERVER_PY = "bin/dev_appserver.py";
   private static final String JAVA_APPENGINE_SDK_PATH =
       "platform/google_appengine/google/appengine/tools/java/lib";
+  private static final String JAVA_COMPONENTS_PATH = "platform/gcd/.appengine";
```

Comment:
Yeah, if you can run `gcloud components list` it would be more reliable/breakage-proof:

    $ gcloud components list --format='value(state)' --filter=id:app-engine-java
    Installed

(Or "Not Installed").

But `.install/google-app-engine.manifest` might be a better file to check the existence of. Mark will have to weigh in on how stable that file is.